app/policies/thredded/post_policy.rb

```diff
@@ -26,7 +26,7 @@ def initialize(user, post)
     end
 
     def create?
-      @user.thredded_admin? || !@post.postable.locked? && messageboard_policy.post?
+      !@user.is_a?(Thredded::NullUser) && (@user.thredded_admin? || !@post.postable.locked?)
```

Comment:
The `NullUser` check is not necessary because `NullUser` also implements the permissions methods.

Please replace with:

```ruby
def create?
  @user.thredded_admin? ||
    !@post.postable.locked? &&
    # Users are allowed to post in unlocked topics of a locked messageboard
    # only if they would be allowed to post if it weren't locked.
    @user.thredded_can_write_messageboards.include?(@post.messageboard)
end
```